datafusion/physical-expr/src/expressions/get_indexed_field.rs

```diff
@@ -107,9 +136,69 @@ impl PhysicalExpr for GetIndexedFieldExpr {
                 }
                 (dt, key) => Err(DataFusionError::NotImplemented(format!("get indexed field is only possible on lists with int64 indexes. Tried {} with {} index", dt, key))),
             },
-            ColumnarValue::Scalar(_) => Err(DataFusionError::NotImplemented(
-                "field access is not yet implemented for scalar values".to_string(),
-            )),
+            ColumnarValue::Scalar(scalar) => match (scalar.get_datatype(), &self.key) {
```

Comment:
I am not sure I fully follow this code -- Since it is creating a `ArrayRef` from the `ColumnarValue::Scalar`, I wonder why it can't use the same code as the `ColumnarValue::Array` case and call `to_arrow()`?

https://github.com/cube-js/arrow-datafusion/blob/scalar-array-index/ballista/rust/client/src/columnar_batch.rs#L150

So for example, rather than 

```rust
    fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
        let arg = self.arg.evaluate(batch)?;
        match arg {
            ColumnarValue::Array(array) => match (array.data_type(), &self.key) {
...
```

It could look like:
```rust
    fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
        let array = self.arg.evaluate(batch)?
          // convert to Arrayref
          .to_arrow();

        match (array.data_type(), &self.key) {
...
```

That way the same code would be used for the array and scalar cases of `ColumnarValue`

Reply:
Thanks, Done.